File: valid-parentheses.py
class Solution:
    def isValid(self, s:str) -> bool:
        stack = []
        mapper = {"{":"}","[":"]","(":")"}
        for x in s:
            if x in mapper:
                print("yes")
                stack.append(mapper[x])
            else:
                if len(stack) != 0:
                    top_element = stack.pop()
                    print(top_element)
                    if x != top_element:
                        return False
                    else:
                        continue
                else:
                    return False
            return len(stack) == 0


isValid(null,"()")

File: javascript/valid-Parentheses.js
var isValid = function(s){
    let valid = true;
    const mapper = {"{":"}","[":"]","(":")"}
    const stack = [];


for(let i in s){
    const v = s[i];
    if(['(','[','{'].indexOf(v) > -1){
        stack.push(v);
    }
    else{
        const peak = stack.pop();
        if(v !== mapper[peak]){
            return false;
        }
    }
}

if(stack.length > 0) return false;

return valid;

};